src/main/java/mezz/jei/ingredients/IngredientListElementComparator.java

```diff
@@ -2,41 +2,281 @@
+    }
+
+    
+    public static String initConfig(/*ConfigTagParent tag*/) {
+        //minecraft, mod, id, default, meta, name
+        //API.addSortOption(
+		addLE("minecraft", new Comparator<IIngredientListElement>() {
+            @Override
+            public int compare(IIngredientListElement o1, IIngredientListElement o2) {
+                boolean m1 = "minecraft".equals(o1.getModName());
+                boolean m2 = "minecraft".equals(o2.getModName());
+                return m1 == m2 ? 0 : m1 ? -1 : 1;
+            }
+        });
+        //API.addSortOption(
+		addLE("mod", new Comparator<IIngredientListElement>() {
+            @Override
+            public int compare(IIngredientListElement o1, IIngredientListElement o2) {
+                String mod1 = o1.getModName();
+                String mod2 = o2.getModName();
+                
+                if (mod1 == null) {
+                    return mod2 == null ? 0 : 1;
+                }
+                if (mod2 == null) {
+                    return -1;
+                }
+                return mod1.compareTo(mod2);
+            }
+        });
+        //API.addSortOption(
+		add("id", new Comparator<ItemStack>() {
+            @Override
+            public int compare(ItemStack o1, ItemStack o2) {
+                int id1 = Item.getIdFromItem(o1.getItem());
+                int id2 = Item.getIdFromItem(o2.getItem());
+                return compareInt(id1, id2);
+            }
+        });
+        //API.addSortOption(
+		addLE("default", new Comparator<IIngredientListElement>() {
+            @Override
+            public int compare(IIngredientListElement o1, IIngredientListElement o2) {
+				final Integer order1 = o1.getOrderIndex();
+				final Integer order2 = o2.getOrderIndex();
+                if (order1 == null) {
+                    return order2 == null ? 0 : 1;
+                }
+                if (order2 == null) {
+                    return -1;
+                }
+                return compareInt(order1, order2);
+            }
+        });
+        //API.addSortOption(
+		add("damage", new Comparator<ItemStack>() {
+            @Override
+            public int compare(ItemStack o1, ItemStack o2) {
+                int id1 = o1.getItemDamage();
+                int id2 = o2.getItemDamage();
+                return compareInt(id1, id2);
+            }
+        });
+        //API.addSortOption
+		addLE("name", new Comparator<IIngredientListElement>() {
+            @Override
+            public int compare(IIngredientListElement o1, IIngredientListElement o2) {
+                String name1 = o1.getDisplayName();
+                String name2 = o2.getDisplayName();
+                return name1.compareTo(name2);
+            }
+        });
+		
+		//Inventory Tweaks needs to be part of the build environment for this bit to work.
+		boolean	hasInvTweaks = false;
+		try {
+			hasInvTweaks = Loader.isModLoaded("inventorytweaks"); 
+		} catch (Exception ex) {
```

Comment:
It does during the gradle build test phase.  I could not find a way to detect the test was running to not call isModLoaded.